pkg/errors/armerrors_test.go

```diff
@@ -135,3 +135,81 @@ func TestRegionalQuotaHasBeenReached(t *testing.T) {
     }
 }
 
+func TestIsNicReservedForAnotherVM(t *testing.T) {
+    testCases := []struct {
+        description   string
+        responseError error
+        expected      bool
+    }{
+        {
+            description: "NIC Reserved for Another VM",
+            responseError: &azcore.ResponseError{
+                ErrorCode:   NicReservedForAnotherVM,
+                StatusCode:  http.StatusForbidden,
+                RawResponse: &http.Response{
+                    Body: io.NopCloser(strings.NewReader(`{"error": {"code": "NicReservedForAnotherVm"}}`)),
+                },
+            },
+            expected: true,
+        },
+        {
+            description: "Different Error Code",
+            responseError: &azcore.ResponseError{
+                ErrorCode:   "ResourceNotFound",
+                StatusCode:  http.StatusNotFound,
+                RawResponse: &http.Response{
+                    Body: io.NopCloser(strings.NewReader(`{"error": {"code": "ResourceNotFound"}}`)),
+                },
+            },
+            expected: false,
+        },
+    }
+
+    for _, tc := range testCases {
+        t.Run(tc.description, func(t *testing.T) {
+            if got := IsNicReservedForAnotherVM(tc.responseError); got != tc.expected {
+                t.Errorf("IsNicReservedForAnotherVM() = %t, want %t for %s", got, tc.expected, tc.description)
+            }
+        })
+    }
+}
+
+func TestIsSKUNotAvailable(t *testing.T) {
```

Comment:
I always use t.Parallel() as first line of my test.

as long as the test have no shared variables/state.